pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1504,6 +1506,16 @@ func CheckNodeDiskPressurePredicate(pod *v1.Pod, meta algorithm.PredicateMetadat
 	return true, nil, nil
 }
 
+// CheckNodePIDPressurePredicate checks if a pod can be scheduled on a node
+// reporting pid pressure condition.
+func CheckNodePIDPressurePredicate(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
```

Comment:
Why didn't you add this check to `CheckNodeConditionPredicate` instead of this function?

Reply:
make sense :)

Reply:
agree with @bsalamat, we might want to check it there as it is nodeCondition.

Also to me, we might want to use taint instead, and let the toleration mechanism do the work, WDYT @k82cn ?

Reply:
@bsalamat  `CheckNodeMemoryPressurePred, CheckNodeDiskPressurePred` is also node condition, why we seprate these to two predicates?

Reply:
@wanghaoran1988 @k82cn I see a benefit to having separate predicates. Users can enable or disable individual predicates in scheduler policy config. So, a user can disable `CheckNodeMemoryPressurePred` and/or `CheckNodeDiskPressurePred` in their policy config while keeping `CheckNodeConditionPredicate` enabled. For the same reason it may make sense to keep `CheckNodePIDPressurePredicate` as a separate predicate as well.

Reply:
yes. separate predicates give user more options :).

> Also to me, we might want to use taint instead, and let the toleration mechanism do the work, 

Handled by https://github.com/kubernetes/kubernetes/pull/60008 .